aiida/manage/tests/pytest_fixtures.py

```diff
@@ -521,6 +525,155 @@ def get_code(entry_point, executable, computer=aiida_localhost, label=None, prep
+        crypto_serialization.PublicFormat.OpenSSH,
+    )
+
+    dirpath = pathlib.Path('~').expanduser() / '.ssh'
+    filename = uuid.uuid4().hex
+    filepath_private_key = dirpath / filename
+    filepath_public_key = dirpath / f'{filename}.pub'
+
+    filepath_private_key.write_bytes(private_key)
+    filepath_public_key.write_bytes(public_key)
+
+    try:
+        yield filepath_private_key
+    finally:
+        filepath_private_key.unlink(missing_ok=True)
+        filepath_public_key.unlink(missing_ok=True)
+
+
+@pytest.fixture
+def aiida_computer(tmp_path) -> t.Callable[[], Computer]:
+    """Factory to return a :class:`aiida.orm.computers.Computer` instance."""
+
+    def factory(
+        label: str = None,
+        minimum_job_poll_interval: int = 0,
+        default_mpiprocs_per_machine: int = 1,
+        configuration_kwargs: dict[t.Any, t.Any] = None,
+        **kwargs
+    ) -> Computer:
+        """Return a :class:`aiida.orm.computers.Computer` instance.
+
+        The database is queried for an existing computer with the given label. If it exists, it means it was probably
+        created by this fixture in a previous call and it is simply returned. Otherwise a new instance is created.
+        Note that the computer is not explicitly configured, unless ``configure_kwargs`` are specified.
+
+        :param label: The computer label. If not specified, a random UUID4 is used.
+        :param minimum_job_poll_interval: The default minimum job poll interval to set.
+        :param configuration_kwargs: Optional keyword arguments that, if defined, are used to configure the computer
+            by calling :meth:`aiida.orm.computers.Computer.configure`.
+        :param kwargs: Optional keyword arguments that are passed to the :class:`aiida.orm.computers.Computer`
+            constructor if a computer with the given label doesn't already exist.
+        :return: A stored computer instance.
+        """
+        label = label or str(uuid.uuid4())
+
+        try:
+            computer = Computer.collection.get(label=label)
+        except NotExistent:
+            computer = Computer(
+                label=label,
+                description=kwargs.pop('description', 'computer created by `aiida_computer` fixture'),
+                hostname=kwargs.pop('hostname', 'localhost'),
+                workdir=kwargs.pop('workdir', str(tmp_path)),
+                transport_type=kwargs.pop('transport_type', 'core.local'),
+                scheduler_type=kwargs.pop('scheduler_type', 'core.direct'),
+            )
+            computer.store()
+            computer.set_minimum_job_poll_interval(minimum_job_poll_interval)
+            computer.set_default_mpiprocs_per_machine(default_mpiprocs_per_machine)
+
+        if configuration_kwargs:
+            computer.configure(**configuration_kwargs)
+
+        return computer
+
+    return factory
+
+
+@pytest.fixture
+def aiida_computer_localhost(aiida_computer) -> t.Callable[[], Computer]:
```

Comment:
Since the label of the computer is configurable, perhaps this could be called `aiida_computer_local`, in parallel to `aiida_computer_ssh`